packages/core/src/odata/common/selectable/collection-field.ts

```diff
@@ -26,9 +27,9 @@ export class CollectionField<
 
   constructor(
     fieldName: string,
-    entityConstructor: Constructable<EntityT>,
+    fieldOf: ConstructorOrField<EntityT>,
```

Comment:
The JSdoc for this states:
```
/**
 * Union type to represent the parent of a complex type field. This can either be an entity constructor or another complex type field.
 */
export type ConstructorOrField<EntityT extends EntityBase, ComplexT> =
```

So it is strange that this is used here in the collection field context. Can you explain?

Reply:
In our Field Hierarchy we have (I abstracted the naming) the BaseField, the EdmTypeFields, the ComplexTypeFields, the EdmTypeComplexTypeFields and the CollectionField. The BaseField is abstract and out of the equasion. The EdmTypeFields are fields of an entity, that have an edm type. EdmTypeComplexTypeFields can only occur as fields of a complex type, so they have the `fieldOf`. The complex type field can occur as a field of a property or another complextype, so it has the `fieldOf`. The same applies to CollectionField. 😊

Reply:
ok fair, but then we should adjust the JSDoc description.

Reply:
Good point, I moved it to a separate file and adjusted the docs.